Tuesdays/Week-Class/Inheratance/Assets/Scripts/TrapBehaviours.cs

```diff
@@ -0,0 +1,30 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class TrapBehaviours{
+	public static void DoDamage(PlayerController player, Trap trap){
+		player.CurrCharacter.TakeDamage(trap.Damage);
+	}
+
+	/// <summary>
+	/// Teleport the player back a random place (from the last 5 that they were at).
+	/// The first element in the stack is the position of that trap, we don't want
+	/// it to count as one of the last 5 places that the player was at. 
+	/// That is why we do one last Pop after the loop.
+	/// </summary>
+	public static void TeleportBack(PlayerController player, Trap trap){
+		int randPlace = Random.Range(0, 5);
+		
+		for(int i = 0; i < randPlace; i++){
+			player.PlayerMovementHistory.Pop();
+		}
```

Comment:
if you don't have five items in the stack and you rolled a number higher then the amount of items you have this will throw an exception.
Plus if you roll 0, the for will never happen